lib/streamlit/commands/logo.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Handle App logos"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from streamlit.elements.image import AtomicImage, WidthBehaviour, image_to_url
+from streamlit.errors import StreamlitAPIException
+from streamlit.proto.ForwardMsg_pb2 import ForwardMsg
+from streamlit.runtime.metrics_util import gather_metrics
+from streamlit.runtime.scriptrunner import get_script_run_ctx
+
+if TYPE_CHECKING:
+    from PIL import Image
+
+
+def _invalid_logo_text(field_name: str):
+    return f"The {field_name} passed to st.logo is invalid - See [documentation](https://docs.streamlit.io/library/api-reference/layout/st.logo) for more information on valid types"
+
+
+@gather_metrics("logo")
+def logo(
+    image: str | Image.Image | AtomicImage,
+    *,  # keyword-only args:
+    link: str | None = None,
+    icon_image: str | Image.Image | AtomicImage | None = None,
+) -> None:
+    """
+    Renders logos in the main and sidebar sections of the page
+
+    Parameters
+    ----------
+    image: numpy.ndarray, BytesIO, or str
```

Comment:
I actually updated `st.chat_message` to align with `st.set_page_config` so it says "Anything supported by `st.image`" instead of trying to tack on long explanation of types. That way, `st.image` can do the heavy lifting and we can just not the special cases that are add-ons. At least, that made sense for those two icon cases...

Reply:
Sounds good will update 👍🏼